packages/markers/src/browser/problem/problem-widget.tsx

```diff
@@ -46,6 +59,44 @@ export class ProblemWidget extends TreeWidget {
         this.addClipboardListener(this.node, 'copy', e => this.handleCopy(e));
     }
 
+    @postConstruct()
+    protected init(): void {
+        super.init();
+        this.updateFollowActiveEditor();
+        this.toDispose.push(this.preferences.onPreferenceChanged(e => {
+            if (e.preferenceName === 'problems.autoReveal') {
+                this.updateFollowActiveEditor();
+            }
+        }));
+    }
+
+    protected updateFollowActiveEditor(): void {
+        this.toDisposeOnCurrentWidgetChanged.dispose();
+        this.toDispose.push(this.toDisposeOnCurrentWidgetChanged);
+        if (this.preferences.get('problems.autoReveal')) {
+            this.followActiveEditor();
+        }
+    }
+
+    protected followActiveEditor(): void {
+        this.autoRevealFromActiveEditor();
+        this.toDisposeOnCurrentWidgetChanged.push(Disposable.create(() => this.autoRevealFromActiveEditor));
+    }
+
+    protected autoRevealFromActiveEditor(): void {
+        this.toDisposeOnCurrentWidgetChanged.push(this.shell.onDidChangeCurrentWidget(e => {
```

Comment:
this line should be removed,  `widget` should be assigned from `shell.currentWidget`